sources/packages/web/src/views/institution/AppInstitution.vue

```diff
@@ -51,6 +53,11 @@ export default {
           });
         }
       }
+      if (isAuthenticated.value) {
+        store.dispatch("institution/initialize");
```

Comment:
Maybe the initialize could be used to call getUserInstitutionDetails and getUserInstitutionLocationDetails in parallel.
Instead of calling 3 dispatch methods here, there will only one dispatch call internally calling the API twice, but in parallel.